src/celeritas/Types.hh

```diff
@@ -105,8 +105,8 @@ enum class TrackStatus : std::int_least8_t
 enum class ActionOrder
 {
     start,  //!< Initialize tracks
-    sort_start,  //!< Sort track slots for GPU optimization
     pre,  //!< Pre-step physics and setup
+    sort_start,  //!< Sort track slots for GPU optimization
```

Comment:
`sort_pre`? since we might be inserting several of these, I think it's better to have them associated with a specific action stage (again until we can come up with a more full-featured dependency graph for the actions).

And if we're sorting by dead/alive we still want to sort immediately after `start` rather than at `pre`, right?

Reply:
Yes for dead/alive we'd ideally be sorting after start though doing it after the pre is OK. Also even though the `SortTracksAction` can take a `TrackOrder` as a constructor argument, in demo-loop we currently don't have a way of specifying different strategy for different `SortTracksActions`

Reply:
OK, we can do that as a follow-on. I'm refactoring some of the demo loop code for the simplecalo/multithread work, so it might be better not to touch it for now.